homeworks/2/internetprogramirane1/src/ClientHandler.java

```diff
@@ -0,0 +1,77 @@
+import java.io.*;
+import java.net.Socket;
+import java.util.ArrayList;
+
+public class ClientHandler implements Runnable {
+
+    public static ArrayList<ClientHandler> clientHandlers = new ArrayList<>();
+    private Socket socket;
+    private BufferedReader bufferedReader;
+    private BufferedWriter bufferedWriter;
+    private String clientUsername;
+
+    public ClientHandler(Socket socket){
+        try{
+            this.socket = socket;
+            this.bufferedWriter = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream()));
+            this.bufferedReader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            this.clientUsername = bufferedReader.readLine();
+            clientHandlers.add(this);
+            broadcastMessage("SERVER: " + clientUsername + " has entered the chat!");
+        }catch (IOException e) {
+            closeEverything(socket, bufferedReader, bufferedWriter);
+        }
+    }
+
+    @Override
+    public void run() {
+        String messageFromClient;
+
+        while(socket.isConnected()) {
+            try {
+                messageFromClient = bufferedReader.readLine();
+                broadcastMessage(messageFromClient);
+            }catch (IOException e){
+                closeEverything(socket, bufferedReader, bufferedWriter);
+                break;
+            }
+        }
+
+    }

```

Comment:
A couple of things here:
1. By the project's requirements there should be functionality that let's the user pick a nickname by sending `/nick <nickname>` to the server. 
2. By the project's requirements there should be functionality that let's the user leave the chat room by sending `/quit`.
3. By the project's requirements the functionality to broadcast messages should only be available when the client's message begins with `/msg`.